src/storage/mvcc/txn.rs

```diff
@@ -101,11 +101,13 @@ impl<'a, T: Engine + ?Sized> MvccTxn<'a, T> {
         }
         // ... or locks at any timestamp.
         if let Some(lock) = meta.get_lock() {
-            return Err(Error::KeyIsLocked {
-                key: key.raw().to_owned(),
-                primary: lock.get_primary_key().to_vec(),
-                ts: lock.get_start_ts(),
-            });
+            if lock.get_start_ts() != self.start_ts {
```

Comment:
Yes. We want to make the prewrite operation idempotent. But I think we should return `Ok(())` here if timestamps are matched. @morefreeze